app/domain/authentication/authn_jwt/jwt_fetch_signing_key_factory.rb

```diff
@@ -0,0 +1,65 @@
+module Authentication
+  module AuthnJwt
+    # Factory that returns the interface implementation of FetchSigningKey
+    class JwtFetchSigningKeyFactory
```

Comment:
the word `Jwt` is redundant. i can live with `FetchJwtSigningKeyFactory` if the signing key is of JWT type